projects/angular/src/data/datagrid/datagrid.ts

```diff
@@ -432,6 +434,16 @@ export class ClrDatagrid<T = any> implements AfterContentInit, AfterViewInit, On
     // we track for changes on selection.current because it can happen with pushing items
     // instead of overriding the variable
     this.selection.checkForChanges();
+
+    if (
+      this.virtualScroll &&
+      this.contentWrapper.nativeElement &&
+      this.isScrollbarVisible !==
+        this.contentWrapper.nativeElement.scrollHeight > this.contentWrapper.nativeElement.clientHeight
+    ) {
+      this.isScrollbarVisible =
+        this.contentWrapper.nativeElement.scrollHeight > this.contentWrapper.nativeElement.clientHeight;
+    }
```

Comment:
Not sure why we need to check the same thing in the `if` condition and then assign the value with the same. Just assigning it when we have virtual scroll should be enough to put the class.
```suggestion

    if (
      this.virtualScroll &&
      this.contentWrapper.nativeElement
    ) {
      this.isScrollbarVisible =
        this.contentWrapper.nativeElement.scrollHeight > this.contentWrapper.nativeElement.clientHeight;
    }
```